### PR TITLE
Cleanup IMU and Image topics

### LINKED ISSUE #196: Cleanup the way depth data gets passed around
Currently we have an `imu_cam_type` that we use for camera images and imu values to be used by SLAM, and then we have a separate `rgb_depth_type` that sends depth images from zed and realsense. The `imu_cam_type` should also contain depth data and set it to null if there isn't a depth camera.

### PR DESCRIPTION
**Pull request by therishidesai**
Closes #196

### CODE REVIEW

**Comment by apache8080** [comment]
After discussing with @mhuzai, I am going to change the way we pass sensor data by having different types/topics for each sensor type (e.g: imu_type for imu data, stereo_gray_type for stereo grayscale images)

**Review by charmoniumQ — DISMISSED** [review]

File: common/data_format.hpp [hunk 1]
```diff
@@ -21,22 +21,33 @@
 namespace ILLIXR {
 
 	typedef std::chrono::time_point<std::chrono::system_clock> time_type;
-	typedef unsigned long long ullong;
+	typedef unsigned long long
```
Comment (charmoniumQ) [inline common/data_format.hpp:24]: This looks like a mistake. Missing a semicolon and a typename (such as ullong)?

File: offline_imu_cam/plugin.cpp [hunk 2]
```diff
@@ -73,20 +74,33 @@ class offline_imu_cam : public ILLIXR::threadloop {
...
 			(sensor_datum.imu0.value().linear_a).cast<float>(),
+			dataset_time,
+		};
+
+		auto stereo_cam_datum = new stereo_img_type{
+			real_now,
 			cam0,
-			cam1,
-			dataset_now,
+			cam1
```
Comment (e3m3) [inline offline_imu_cam/plugin.cpp:98]:
```suggestion
			cam1,
```

File: realsense/plugin.cpp [hunk 3]
```diff
@@ -113,36 +113,33 @@ class realsense : public plugin {
                     // Images
                     std::optional<cv::Mat *> img0 = std::nullopt;
                     std::optional<cv::Mat *> img1 = std::nullopt;
-                    std::optional<cv::Mat *> rgb = std::nullopt;
-                    std::optional<cv::Mat *> depth = std::nullopt;
+					std::optional<cv::Mat *> rgb_img0 = std::nullopt;
+					// Realsense only has rgb image for left image so rgb_img1 will always be null
+                    std::optional<cv::Mat *> rgb_img1 = std::nullopt;
+                    std::optional<cv::Mat *> depth_img = std::nullopt;
```
Comment (e3m3) [inline realsense/plugin.cpp:119]:
```suggestion
                    std::optional<cv::Mat *> rgb_img0 = std::nullopt;
                    // Realsense only has rgb image for left image so rgb_img1 will always be null
                    std::optional<cv::Mat *> rgb_img1 = std::nullopt;
                    std::optional<cv::Mat *> depth_img = std::nullopt;
```

File: realsense/plugin.cpp [hunk 4]
```diff
@@ -113,36 +113,33 @@ class realsense : public plugin {
...
+					
                     if (last_iteration_cam != cam_type_.iteration)
                     {
                         last_iteration_cam = cam_type_.iteration;
                         img0 = cam_type_.img0;
                         img1 = cam_type_.img1;
-                        rgb = cam_type_.rgb;
-                        depth = cam_type_.depth;
+						rgb_img0 = cam_type_rgb_img0;
+                        depth_img = cam_type_.depth;
```
Comment (e3m3) [inline realsense/plugin.cpp:127]:
```suggestion
                        gb_img0 = cam_type_rgb_img0;
                        depth_img = cam_type_.depth;
```

File: zed/src/main.cpp [hunk 5]
```diff
@@ -21,7 +21,8 @@ const record_header __imu_cam_record {"imu_cam", {
 typedef struct {
     cv::Mat* img0;
     cv::Mat* img1;
-    cv::Mat* rgb;
+    cv::Mat* rgb_img0;
+	cv::Mat* rgb_img1;
```
Comment (e3m3) [inline zed/src/main.cpp:25]:
```suggestion
    cv::Mat* rgb_img0;
    cv::Mat* rgb_img1;
```

File: zed/src/main.cpp [hunk 6]
```diff
@@ -64,12 +65,14 @@ class zed_camera_thread : public threadloop {
...
-        rgb_zed.alloc(image_size.width, image_size.height, MAT_TYPE::U8_C4, MEM::CPU);
+        rgbL_zed.alloc(image_size.width, image_size.height, MAT_TYPE::U8_C4, MEM::CPU);
+        rgbR_zed.alloc(image_size.width, image_size.height, MAT_TYPE::U8_C4, MEM::CPU);
         depth_zed.alloc(image_size.width, image_size.height, MAT_TYPE::F32_C1, MEM::CPU);
 
         imageL_ocv = slMat2cvMat(imageL_zed);
         imageR_ocv = slMat2cvMat(imageR_zed);
-        rgb_ocv = slMat2cvMat(rgb_zed);
+		rgbL_ocv = slMat2cvMat(rgbL_zed);
+		rgbR_ocv = slMat2cvMat(rgbR_zed);
```
Comment (e3m3) [inline zed/src/main.cpp:75]:
```suggestion
       rgbL_ocv = slMat2cvMat(rgbL_zed);
       rgbR_ocv = slMat2cvMat(rgbR_zed);
```

File: zed/src/main.cpp [hunk 7]
```diff
@@ -84,12 +87,14 @@ class zed_camera_thread : public threadloop {
     Mat imageL_zed;
     Mat imageR_zed;
     Mat depth_zed;
-    Mat rgb_zed;
+    Mat rgbL_zed;
+	Mat rgbR_zed;
```
Comment (e3m3) [inline zed/src/main.cpp:91]:
```suggestion
    Mat rgbL_zed;
    Mat rgbR_zed;
```

File: zed/src/main.cpp [hunk 8]
```diff
@@ -84,12 +87,14 @@ class zed_camera_thread : public threadloop {
...
 
     cv::Mat imageL_ocv;
     cv::Mat imageR_ocv;
     cv::Mat depth_ocv;
-    cv::Mat rgb_ocv;
+    cv::Mat rgbL_ocv;
+	cv::Mat rgbR_ocv;
```
Comment (e3m3) [inline zed/src/main.cpp:97]:
```suggestion
    cv::Mat rgbL_ocv;
    cv::Mat rgbR_ocv;
```

File: zed/src/main.cpp [hunk 9]
```diff
@@ -105,7 +110,8 @@ class zed_camera_thread : public threadloop {
         zedm->retrieveImage(imageL_zed, VIEW::LEFT_GRAY, MEM::CPU, image_size);
         zedm->retrieveImage(imageR_zed, VIEW::RIGHT_GRAY, MEM::CPU, image_size);
         zedm->retrieveMeasure(depth_zed, MEASURE::DEPTH, MEM::CPU, image_size);
-        zedm->retrieveImage(rgb_zed, VIEW::LEFT, MEM::CPU, image_size);
+        zedm->retrieveImage(rgbL_zed, VIEW::LEFT, MEM::CPU, image_size);
+		zedm->retrieveImage(rgbR_zed, VIEW::RIGHT, MEM::CPU, image_size);
```
Comment (e3m3) [inline zed/src/main.cpp:114]:
```suggestion
        zedm->retrieveImage(rgbL_zed, VIEW::LEFT, MEM::CPU, image_size);
        zedm->retrieveImage(rgbR_zed, VIEW::RIGHT, MEM::CPU, image_size);
```

File: zed/src/main.cpp [hunk 10]
```diff
@@ -175,16 +182,18 @@ class zed_imu_thread : public threadloop {
 
         std::optional<cv::Mat*> img0 = std::nullopt;
         std::optional<cv::Mat*> img1 = std::nullopt;
-        cv::Mat* depth = nullptr;
-        cv::Mat* rgb = nullptr;
+        std::optional<cv::Mat*> rgb_img0 = std::nullopt;
+        std::optional<cv::Mat*> rgb_img1 = std::nullopt;
+		std::optional<cv::Mat*> depth_img = std::nullopt;
```
Comment (e3m3) [inline zed/src/main.cpp:187]:
```suggestion
        std::optional<cv::Mat*> rgb_img0 = std::nullopt;
        std::optional<cv::Mat*> rgb_img1 = std::nullopt;
        std::optional<cv::Mat*> depth_img = std::nullopt;
```

File: zed/src/main.cpp [hunk 11]
```diff
@@ -175,16 +182,18 @@ class zed_imu_thread : public threadloop {
...
         const cam_type* c = _m_cam_type->get_latest_ro();
         if (c && c->serial_no != last_serial_no) {
             last_serial_no = c->serial_no;
             img0 = c->img0;
             img1 = c->img1;
-            depth = c->depth;
-            rgb = c->rgb;
+			rgb_img0 = c->rgb_img0;
+            rgb_img1 = c->rgb_img1;
+            depth_img = c->depth;
```
Comment (e3m3) [inline zed/src/main.cpp:196]:
```suggestion
            gb_img0 = c->rgb_img0;
            rgb_img1 = c->rgb_img1;
            depth_img = c->depth;
```

File: zed/src/main.cpp [hunk 12]
```diff
@@ -193,21 +202,17 @@ class zed_imu_thread : public threadloop {
...
+				imu_time_point,
+					av,
+					la,
+					img0,
+					img1,
+					rgb_img0,
+					rgb_img1,
+					depth_img,
+					imu_time,
         });
```
Comment (e3m3) [inline zed/src/main.cpp:214]:
```suggestion
                imu_time_point,
                av,
                la,
                img0,
                img1,
                rgb_img0,
                rgb_img1,
                depth_img,
                imu_time,
        });
```

**Review by e3m3 — CHANGES_REQUESTED** [review]
Looks good other than the issue @charmoniumQ pointed out, and the provided formatting suggestion

**Comment by mhuzai** [comment]
This is quite out-of-date now, and should get tackled afresh in preparation for the interfaces working group.

**Pull request closed without merging**